### PR TITLE
Linting

### PR DESCRIPTION
**Pull request by csutorasr**
I have added pylint to the project, and corrected many linting errors on the algorithms connected to arrays.

### CODE REVIEW

File: array/garage.py [hunk 1]
```diff
@@ -24,7 +24,7 @@
 initial: [1, 2, 3, 0, 4]
 final:   [0, 3, 2, 1, 4]
 Steps =  4
-Sequence : 
+Sequence :
```
Comment (szaszm) [inline array/garage.py]: The space is not needed before the colon.

File: array/merge_intervals.py [hunk 2]
```diff
@@ -61,7 +61,7 @@ def print_intervals(intervals):
         res = []
         for i in intervals:
             res.append(repr(i))
-        print("".join(res))
+        print "".join(res)
```
Comment (szaszm) [inline array/merge_intervals.py]: This broke python3 compatibility, so please revert this change

File: array/move_zeros_to_end.py [hunk 3]
```diff
@@ -17,6 +18,6 @@ def move_zeros(array):
             zeros += 1
         else:
             result.append(i)
-    
+
```
Comment (szaszm) [inline array/move_zeros_to_end.py]: Does it work without the indentation? In python indentation is part of the language syntax and altering it can change program behaviour.

File: array/plus_one.py [hunk 4]
```diff
@@ -40,8 +43,11 @@ def plus_one_v2(digits):
 
 
 def plus_one_v3(num_arr):
+    """
+    :type num_arr: List[int]
+    """
 
-    for idx, digit in reversed(list(enumerate(num_arr))):
+    for idx, in reversed(list(enumerate(num_arr))):
```
Comment (szaszm) [inline array/plus_one.py]: Isn't this a syntax error?

Comment (szaszm) [reply array/plus_one.py]: If not, consider using underscore for the discarded loop variable.

File: array/test_array.py [hunk 5]
```diff
@@ -117,7 +117,8 @@ def test_garage(self):
         steps, seq = garage(initial, final)
 
         self.assertEqual(steps, 4)
-        self.assertListEqual(seq, [[0, 2, 3, 1, 4], [2, 0, 3, 1, 4], [2, 3, 0, 1, 4], [0, 3, 2, 1, 4]])
+        self.assertListEqual(seq,
+                             [[0, 2, 3, 1, 4], [2, 0, 3, 1, 4], [2, 3, 0, 1, 4], [0, 3, 2, 1, 4]])
```
Comment (szaszm) [inline array/test_array.py]: This did not help readibility at all.

File: array/move_zeros_to_end.py (hunk 3 above)
Comment (csutorasr) [reply array/move_zeros_to_end.py]: Yes.

File: array/test_array.py (hunk 5 above)
Comment (csutorasr) [reply array/test_array.py]: But the line is not longer than 100 characters.

File: array/move_zeros_to_end.py (hunk 3 above)
Comment (csutorasr) [reply array/move_zeros_to_end.py]: As you can see the tests have run.

File: array/test_array.py (hunk 5 above)
Comment (szaszm) [reply array/test_array.py]: what about something like this? may need backslashes at the end of the lines.
[
    [0, 2, 3, 1, 4],
    [2, 0, 3, 1, 4],
    [2, 3, 0, 1, 4],
    [0, 3, 2, 1, 4]
]

File: array/move_zeros_to_end.py (hunk 3 above)
Comment (thomasklein94) [reply array/move_zeros_to_end.py]: AFAIK blank lines are disregarded by the python interpreter, and i think its makes sense to not have a randomly idented empty line in the code. Althrough PEP8 says nothing about this matter (https://www.python.org/dev/peps/pep-0008/#blank-lines)

File: array/test_array.py (hunk 5 above)
Comment (csutorasr) [reply array/test_array.py]: I like it but it shows 5 linting error. But I made the changes.

Comment (thomasklein94) [reply array/test_array.py]: Can we expand the sub-arrays as well? So it would be look like
```python
self.assertListEqual(
                                seq,
                                [
                                    [0, 2, 3, 1,4],
# and so on
```

File: array/three_sum.py [hunk 6]
```diff
@@ -25,24 +25,24 @@ def three_sum(array):
...
-        l, r = i + 1, len(array) - 1
-        while l < r:
-            s = array[i] + array[l] + array[r]
-            if s > 0:
-                r -= 1
-            elif s < 0:
-                l += 1
+        left, right = i + 1, len(array) - 1
+        while left < right:
+            summa = array[i] + array[left] + array[right]
```
Comment (thomasklein94) [inline array/three_sum.py]: Can you change `summa` to `sum`? English does not have a word summa, and it would be nice if all variables are in english.
(to see how bad it can be sometimes, there is a funny counterexample in php: [paamayim-nekudotayim](http://php.net/manual/en/language.oop5.paamayim-nekudotayim.php) )

Comment (csutorasr) [reply array/three_sum.py]: `sum` is a built-in as you can see in the commit message. I changed it to latin. If you have any better idea change it in a commit.

File: array/move_zeros_to_end.py (hunk 3 above)
Comment (csutorasr) [reply array/move_zeros_to_end.py]: I just removed the trailing spaces. Feel free to remove the line, but it is not a linting error.

File: array/three_sum.py (hunk 6 above)
Comment (thomasklein94) [reply array/three_sum.py]: You are absolutely right. in this case i would go with `sum_`.

File: README.md [hunk 7]
```diff
@@ -12,6 +12,20 @@ Minimal and clean example implementations of data structures and algorithms in P
 Thanks for your interest in contributing! There are many ways to contribute to this project. [Get started here](CONTRIBUTING.md)
 
 
+## Linting
```
Comment (ducktin) [inline README.md]: It would be a good idea to introduce package level linting by adding __init__.py to each subdirectory. By doing this there would be an option to run:
```bash
pylint ./packagename
```

Comment (csutorasr) [reply README.md]: Create a PR for that.

Comment (ducktin) [reply README.md]: Let's discuss this in a separate issue.
